curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo/64ec8f717b261e824d82d6a5.md

```diff
@@ -0,0 +1,332 @@
+---
+id: 64ec8f717b261e824d82d6a5
+title: Step 15
+challengeType: 0
+dashedName: step-15
+---
+
+# --description--
+
+Now that you have the input fields' values and an `id`, you want to add them to your `taskData` array to keep track of each task. Remember you would only do that if the task is new, otherwise, you will set it up for editing. That's why you have the `dataArrIndex` variable to get the index of each task.
+
+Create an `if` statement and make the condition `dataArrIndex === -1`. Inside the `if` statement, use `unshift()` to add the `taskObj` object to the `taskData` array.
+
+`unshift()` is an array method that is used to add one or more elements to the beginning of an array.
```

Comment:
```suggestion
```

Reply:
I have this here because `unshift` might not have been introduced at this point in the curriculum.

Reply:
What do you think?

Reply:
What do you think?